### PR TITLE
Blame Sequence File Changes

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
## Blame Sequence File Changes
Added Display name and Is Test Completed in the sequence file that is generated by BlameDataCollector

### CODE REVIEW

**Comment by vagisha-nidhi** [comment]
@kaadhina Please review these changes

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameTestObject.cs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        #endregion
+
+        #region Private Methods
+
+        /// <summary>
+        /// Creates a Id of TestCase
+        /// </summary>
+        /// <returns>Guid test id</returns>
+        private Guid GetTestId()
+        {
```
Comment (kaadhina) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameTestObject.cs]: remove

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs [hunk 2]
```diff
@@ -155,7 +155,7 @@ private void EventsTestCaseStart(object sender, TestCaseStartEventArgs e)
                 EqtTrace.Info("Blame Collector : Test Case Start");
             }
 
-            this.testSequence.Add(e.TestElement);
+            this.testSequence.Add(new BlameTestObject(e.TestElement));
             this.testStartCount++;
```
Comment (kaadhina) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs]: make dict

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/XmlReaderWriter.cs [hunk 3]
```diff
@@ -55,7 +55,7 @@ protected XmlReaderWriter(IFileHelper fileHelper)
         /// The file Path.
         /// </param>
         /// <returns>File path</returns>
-        public string WriteTestSequence(List<TestCase> testSequence, string filePath)
```
Comment (kaadhina) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/XmlReaderWriter.cs]: >TestCase [](start = 45, length = 8)

check for outcome/result

File: test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/BlameCollectorTests.cs [hunk 4]
```diff
@@ -198,7 +198,7 @@ public void TriggerSessionEndedHandlerShouldGetDumpFileIfCollectDumpOnExitIsEnab
 
             // Setup
             this.mockProcessDumpUtility.Setup(x => x.GetDumpFile()).Returns(this.filepath);
-            this.mockBlameReaderWriter.Setup(x => x.WriteTestSequence(It.IsAny<List<TestCase>>(), It.IsAny<string>()))
+            this.mockBlameReaderWriter.Setup(x => x.WriteTestSequence(It.IsAny<List<BlameTestObject>>(), It.IsAny<string>()))
```
Comment (kaadhina) [inline test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/BlameCollectorTests.cs]: UTs for both completed and not completed

**Comment by kaadhina** [comment]
>             this.mockFileHelper.Setup(m => m.Exists(It.IsAny<string>())).Returns(true);

Add validations for the new attributes

---
Refers to: test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/XmlReaderWriterTests.cs:107 in 353998e. [](commit_id = 353998e49883efc9cf13e44c93dca90c5d90e67d, deletion_comment = False)

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameTestObject.cs (hunk 1 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameTestObject.cs]: Removed this

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs (hunk 2 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs]: Dictionary and list used.

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/XmlReaderWriter.cs (hunk 3 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Extensions.BlameDataCollector/XmlReaderWriter.cs]: There is no outcome field in TestCase object.

File: test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/BlameCollectorTests.cs (hunk 4 above)
Comment (vagisha-nidhi) [reply test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/BlameCollectorTests.cs]: Added UTs in XmlReaderWriterTests.cs

File: test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/XmlReaderWriterTests.cs [hunk 5]
```diff
@@ -23,8 +23,9 @@ public class XmlReaderWriterTests
         private TestableXmlReaderWriter xmlReaderWriter;
         private Mock<IFileHelper> mockFileHelper;
         private Mock<Stream> mockStream;
-        private List<TestCase> testCaseList;
-        private TestCase testcase;
+        private List<Guid> testCaseList;
+        private Dictionary<Guid, BlameTestObject> testObjectDictionary;
```
Comment (kaadhina) [inline test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/XmlReaderWriterTests.cs]: nit: rename to something like testsequence

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs [hunk 6]
```diff
@@ -155,7 +157,15 @@ private void EventsTestCaseStart(object sender, TestCaseStartEventArgs e)
...
+            var blameTestObject = new BlameTestObject(e.TestElement);
+
+            // Add guid to list of test sequence to maintain the order.
+            this.testSequence.Add(blameTestObject.Id);
+
+            // Add the test object to the dictionary.
+            this.testObjectDictionary.Add(blameTestObject.Id, blameTestObject);
+
+            // Increment test start count.
             this.testStartCount++;
```
Comment (kaadhina) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs]: >this.testStartCount++; [](start = 12, length = 22)

Do we still need this>?

File: test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/XmlReaderWriterTests.cs [hunk 7]
```diff
@@ -138,6 +143,59 @@ public void WriteTestSequenceShouldWriteFileStream()
...
+            var testObjectDictionary = new Dictionary<Guid, BlameTestObject>
+            {
+                { testObject.Id, testObject }
+            };
+
+            var filePath = xmlReaderWriter.WriteTestSequence(testSequence, testObjectDictionary, Path.GetTempPath());
+            var testCaseList = xmlReaderWriter.ReadTestSequence(filePath);
+            File.Delete(filePath);
+
+            Assert.AreEqual(testCaseList.First().FullyQualifiedName, "Abc.UnitTest1");
```
Comment (kaadhina) [inline test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/XmlReaderWriterTests.cs]: Displayname check?

Comment (vagisha-nidhi) [reply test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/XmlReaderWriterTests.cs]: Will add that.

File: test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/BlameCollectorTests.cs [hunk 8]
```diff
@@ -91,14 +91,14 @@ public void TriggerSessionEndedHandlerShouldWriteToFileIfTestHostCrash()
...
+            this.mockBlameReaderWriter.Setup(x => x.WriteTestSequence(It.IsAny<List<Guid>>(), It.IsAny<Dictionary<Guid, BlameTestObject>>(), It.IsAny<string>()))
                 .Returns(this.filepath);
 
             this.mockDataColectionEvents.Raise(x => x.TestCaseStart += null, new TestCaseStartEventArgs(testcase));
             this.mockDataColectionEvents.Raise(x => x.SessionEnd += null, new SessionEndEventArgs(this.dataCollectionContext));
 
             // Verify WriteTestSequence Call
-            this.mockBlameReaderWriter.Verify(x => x.WriteTestSequence(It.IsAny<List<TestCase>>(), It.IsAny<string>()), Times.Once);
+            this.mockBlameReaderWriter.Verify(x => x.WriteTestSequence(It.IsAny<List<Guid>>(), It.IsAny<Dictionary<Guid, BlameTestObject>>(), It.IsAny<string>()), Times.Once);
         }
```
Comment (kaadhina) [inline test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/BlameCollectorTests.cs]: validation for values in the dictionary and list, we should validate that iscompleted, fqdn, display name etc were set properly

**Review by kaadhina — APPROVED** [review]
:shipit:

**Comment by vagisha-nidhi** [comment]
@dotnet-bot test this please

**Pull request merged**